classify_genes.R

```diff
@@ -78,6 +82,16 @@ num_groups = length(groups_to_keep) ## count how many groups there are
 
 if (num_groups < 2) { stop("Not enough groups to compared with at least min_size isolates! Try decreasing min_size to compare smaller groups.") }
 
+## deal with singleton genes
+if (remove_singletons)  {
+  num_singletons = length(unlist(complete_presence_absence[rowSums(complete_presence_absence[,2:ncol(complete_presence_absence)])==1,1])) ## count singletons
```

Comment:
there's a bit of code duplication here and redefining the same expression for getting singletons multiple times, ideally you would only do that once, could you simply use:
```
singleton_genes = as.character(unlist(complete_presence_absence[rowSums(complete_presence_absence[,2:ncol(complete_presence_absence)])==1,1]))
num_singletons = length(singleton_genes)
if (length(singleton_genes) > 0) {
# remove singletons here
}
```

Reply:
Yes, very true, will change. Although I think the if statement would need to be as below, where "remove_singletons" is determined in the options
```
if (remove_singletons) {
# remove singletons, report number removed
}
```